vite-project/src/strategypage/RadarChart.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useRef, useEffect } from "react";
+
+export interface point {
+  x: number;
+  y: number;
+}
+
+export interface RadarInput {
+  name: string;
+  value: number;
+  max: number;
+}
+
+export interface RadarComponentProps {
+  inputs: RadarInput[];
+}
+
+export const RadarComponent: React.FC<RadarComponentProps> = ({ inputs }) => {
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+
+  const polygonPoints: point[] = new Array(inputs.length);
+  const statPoints: point[] = new Array(inputs.length);
+
+  //function that draws the outer regular polygos
+
+  function drawOuterPolygon(
+    polygonRadius: number,
+    canvas: HTMLCanvasElement,
+    polygonPoints: point[],
+    stokeColor: string
+  ): void {
+    const context = canvas.getContext("2d");
+    const canvasRadius: number = canvas.width / 2;
+    if (!context) {
+      return;
+    }
+    context.beginPath();
+
+    for (let i = 0; i < inputs.length + 1; i++) {
+      const currentX: number =
+        polygonRadius * Math.sin((i * Math.PI * 2) / inputs.length) +
+        canvasRadius;
+      const currentY: number =
+        polygonRadius * Math.cos((i * Math.PI * 2) / inputs.length) +
+        canvasRadius;
```

Comment:
```suggestion
      const [x, y] = coordsFromAngle(polygonWidth, i / inputs.length); // function coordsFromAngle(radius: number, fractionAroundCircle: number): [number, number] (and can be defined outside `RadarComponent`
```

Reply:
I think because it is a very specific case and it is already inside a function there is no use for this function